extensions/src/doodlebot/Doodlebot.ts

```diff
@@ -645,83 +645,202 @@ export default class Doodlebot {
     line;
     lineCounter = 0;
     detector;
-
+    j = 0;
```

Comment:
I know this will be tedious, but can we specify every variable? I assume `j` is an iterator?